torch/fx/graph_module.py

```diff
@@ -546,7 +547,7 @@ def __reduce_deploy__(self, importer: Importer, tracer_cls: Type = None):
         del dict_without_graph['_graph']
         return (reduce_deploy_graph_module, (dict_without_graph, import_block, tracer_cls))
 
-    def __reduce_package__(self, exporter: PackageExporter):
+    def __reduce_package__(self, exporter: PackageExporter, tracer_cls: Type = None):
```

Comment:
Hmm, I should have said this on the reduce_deploy PR but seeing this again I realize: my preference is to add `tracer_cls` to the state of GraphModule rather than thread it through the reduce protocol as we are doing here. Two reasons for this:
1. The reduction function arguments are BC sensitive, so I'd prefer not to add new ones (to be clear, this is a worthy use case for it, but if there's an alternative maybe we should do that).
2. The current implementation requires that an inherited GraphModule class be defined so that we can curry the arguments together (in this case it is `DperGraphModule`) . If we make it automatically the case that all GraphModules store their tracer cls, then all users would benefit without having to do anything I think.

What do you think @bradleyhd @HarutMov?

Reply:
Looks like this approach was already discussed here : https://github.com/pytorch/pytorch/issues/54573. At that time it was not possible to pickle Tracer object, we need to check whether it is still a blocker.

Reply:
we're not pickling the tracer object though, just the class, which should be fine to pickle